module/VuFind/src/VuFind/Marc/Serialization/SerializationFileAbstractBase.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Abstract base class for serialization format support classes.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  MARC
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+namespace VuFind\Marc\Serialization;
+
+/**
+ * Abstract base class for serialization format support classes.
+ *
+ * @category VuFind
+ * @package  MARC
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+abstract class SerializationFileAbstractBase implements SerializationFileInterface
+{
+    /**
+     * Message callback
+     *
+     * @var callable
+     */
+    protected $messageCallback = null;
+
+    /**
+     * Check if the serialization class can parse the given MARC collection file
+     *
+     * @param string $file File name
+     *
+     * @return bool
+     */
+    abstract public static function canParseCollectionFile(string $file): bool;
```

Comment:
I might be mistaken, but I don't think you need to add abstract function declarations for interface methods in an abstract class unless you really want to.

Reply:
You're not mistaken, and I removed the declarations. Thanks, so much better. :)